### PR TITLE
[Feature:UI/UX] user profile settings page

### LINKED ISSUE #5515: Introduce a user settings page
**What problem are you trying to solve with Submitty**
When a user first logs into submitty they are greeted with the home page.  The `About You` section is starting to become cluttered up with several new options that are configured by the user.  These are all settings that configured on a per-user basis instead of on a per-course basis like many of the other settings in submitty.

**Describe the way you'd like to solve this problem**
- [ ] Introduce a new `Settings` page which will be available via a button on the left hand navigation bar from the home page.
- [ ] Allow the user to configure their preferred first and last name from the `Settings` page
- [ ] Allow the user to configure their time zone from the `Settings` page
- [ ] Allow the user to configure their user image from the `Settings` page
- [ ] Allow the user to configure their color theme from the `Settings` page
- [ ] Remove the popup modal from the `About You` section of the home page.  (Do leave the `About You` area so users can see their settings at a glance, however to edit any settings they now need to go to the `Settings` page.)

### PR DESCRIPTION
**Pull request by mukul-kmr-jha**
### Please check if the PR fulfills these requirements:

* [x] The PR title and message follows our [guidelines](https://submitty.org/developer/how_to_contribute)
* [x] Tests for the changes have been added/updated (if possible)
* [x] Documentation has been updated/added if relevant

### What is the current behavior?
Currently, user profile info and the courses in which s/he is enrolled are shown on the home page.

### What is the new behavior?
Closes #5515 

Adds a separate space for the user profile information where they can view and update their various information (all Ajaxified, no page reloads!) like the basic info, profile photo, time-zone, and theme to use. Also, a separate page has been added to access all the archived and current courses in which they have participated in any form (student, instructor, or any type of grader). These pages can be accessed via the new menu options provided in the side navigation bar. 

Also, a single time-zone drop-down list is added with the UTC offset value attached to it. This list is sorted in increasing order wrt UTC offset values to let users find the best-suited time-zone quickly. (changes requested by @MasterOdin in this [comment](https://github.com/Submitty/Submitty/pull/5400#issuecomment-633136302).) 
 
image

### Other information?
~Might need to bring some more changes in the home page.
My courses page is WIP~

### CODE REVIEW

**Comment by jportorreal00** [comment]
Hey @mukul-kmr-jha, this looks really good! The error checking is good. I saw that you said it would have no page reloads. Does this mean that once updated the user should see the changes? If, so I wasn't able to have the profile picture updated without reloading the page.

**Comment by mukul-kmr-jha** [comment]
Hii @jportorreal00 class name was updated for the profile-photo div leading to that bug... anyways fixed now.. please take another look, thanks!

File: site/app/views/UserProfileView.php [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+     * @param bool $database_authentication
+     * @param string $csrf_token
+     * @return string
+     */
+    public function showUserProfile(
+        User $user,
+        string $change_name_text,
+        bool $database_authentication,
+        string $csrf_token
+    ) {
```
Comment (Kevin-MM) [inline site/app/views/UserProfileView.php:25]: I'm not sure how I feel about splitting this onto multiple lines.  It does increase readability but it doesn't seem very consistent with the rest of the code base.  I would like to solicit additional feedback from @bmcutler and @MasterOdin .

File: site/public/css/homepage.css [hunk 2]
```diff
@@ -62,3 +62,10 @@ select {
 #user_settings_modal input[type='file'] {
     margin-top: 10px;
 }
+
+/*.flex-wrapper .content {*/
+/*    flex: 1;*/
+/*}*/
+/*.flex-wrapper .content:nth-child(2) {*/
+/*    margin-left: 0;*/
+/*}*/
```
Comment (Kevin-MM) [inline site/public/css/homepage.css:71]: Are these needed?

Comment (mukul-kmr-jha) [reply site/public/css/homepage.css]: Nope, removed

**Comment by mukul-kmr-jha** [comment]
Fixed this flag message issue and about this 'my courses' page I actually added this view-only information for the user on home-page along with courses but later we decided to create a different page for 'my-courses' because there was suggestion to remove the 'About you' section from the home page and to just keep a simple hello message, like "hello, Quinn".

**Pull request merged**